### PR TITLE
core/vm: set tracer-observable value of a delegatecall to match parent value

### PR DESCRIPTION
**Pull request by AskAlexSharov**
requested by https://github.com/ledgerwatch/erigon/issues/9549

port of https://github.com/ethereum/go-ethereum/pull/26632

### CODE REVIEW

**Comment by racytech** [comment]
I'll take a look at it

**Comment by AskAlexSharov** [comment]
@racytech ping

File: eth/stagedsync/stage_log_index_test.go [hunk 1]
```diff
@@ -92,34 +91,6 @@ func genReceipts(t *testing.T, tx kv.RwTx, blocks uint64) (map[libcommon.Address
 	return expectAddrs, expectTopics
 }
 
-func TestPromoteLogIndex(t *testing.T) {
```
Comment (yperbasis) [inline eth/stagedsync/stage_log_index_test.go:95]: why remove this test?

File: eth/stagedsync/stage_log_index_test.go [hunk 2]
```diff
@@ -168,37 +139,3 @@ func TestPruneLogIndex(t *testing.T) {
 		require.Equal(total, 60) // 1/3rd of 45 not pruned as it has address "1", so 30 Pruned in total, remaining 90-30
 	}
 }
-
-func TestUnwindLogIndex(t *testing.T) {
```
Comment (yperbasis) [inline eth/stagedsync/stage_log_index_test.go:172]: ditto, why remove this?

File: eth/stagedsync/stage_log_index_test.go (hunk 1 above)
Comment (AskAlexSharov) [reply eth/stagedsync/stage_log_index_test.go:95]: probably unrelated

**Review by yperbasis — APPROVED** [review]

**Pull request merged**